packages/pds/src/api/com/atproto/data.ts

```diff
@@ -0,0 +1,27 @@
+import { Server } from '../../../lexicon'
+import * as locals from '../../../locals'
+import * as repo from '../../../repo'
+import ServerAuth from '../../../auth'
+
+export default function (server: Server) {
+  server.com.atproto.data.uploadFile({
+    auth: ServerAuth.verifier,
+    handler: async ({ input, req, res }) => {
+      const { db, blobstore } = locals.get(res)
+
+      const cid = await repo.addUntetheredBlobStream(
+        db,
+        blobstore,
+        input.encoding,
+        req,
```

Comment:
the bytes don't get read off the req in middleware, so we can just send that readstream directly to the blob processor.

We may want to make this more explicit in xrpc-server, & pipe them into a stream that we stick on to `input`. Thoughts?

Reply:
Yepyep, that sounds good to me.